sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/it/SpannerChangeStreamIT.java

```diff
@@ -93,6 +101,22 @@ public void before() {
 
   @Test
   public void testReadSpannerChangeStream() {
+    testReadSpannerChangeStreamImpl(pipeline, null);
+  }
+
+  @Test
+  public void testReadSpannerChangeStreamWithAuthorizedRole() {
+    testReadSpannerChangeStreamImpl(pipeline, ENV.getDatabaseRole());
+  }
+
+  @Test
+  public void testReadSpannerChangeStreamWithUnauthorizedRole() {
+    exception.expect(SpannerException.class);
```

Comment:
the added tests will be executed on both direct runner precommit and dataflow postcommit:

https://ci-beam.apache.org/view/PostCommit/job/beam_PreCommit_Java_GCP_IO_Direct_Cron

https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_Java_DataflowV2

For the latter the exception would not thrown. If remote pipeline fails the PipelineResult is set to FAIL.

Reply:
If we only need to validate the error handling on direct runner we can add something like `assumeTrue(options.getRunner() == DirectRunner.class);`

Reply:
Thanks for pointing this out! I've added the assumeTrue statement.